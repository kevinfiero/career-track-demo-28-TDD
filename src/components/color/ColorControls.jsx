import React from 'react';
import PropTypes from 'prop-types';

const ColorControls = ({ bgColor, fgColor, text, onChange }) => (
  <>
    <label htmlFor="bgColor">Background Color</label>
    <input 
      id="bgColor" 
      name="bgColor"
      type="color" 
      value={bgColor} 
      onChange={onChange} 
    />
    <label htmlFor="fgColor">Foreground Color</label>
    <input 
      id="fgColor"
      name="fgColor"
      type="color" 
      value={fgColor} 
      onChange={onChange} 
    />
    <label htmlFor="text">Text</label>
    <input 
      id="text"
      name="text"
      type="text" 
      value={text} 
      onChange={onChange} 
    />
  </>
);

ColorControls.propTypes = {
  bgColor: PropTypes.string.isRequired,
  fgColor: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default ColorControls;
