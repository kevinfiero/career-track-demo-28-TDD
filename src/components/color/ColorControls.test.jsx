import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ColorControls from './ColorControls';

describe('ColorControls component', () => {
  afterEach(() => cleanup());
  it('renders ColorControls', () => {
    const { asFragment } = render(<ColorControls  
      bgColor="#FF0000" 
      fgColor="#0000FF" 
      text="Hello!" 
      onChange={jest.fn()}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
