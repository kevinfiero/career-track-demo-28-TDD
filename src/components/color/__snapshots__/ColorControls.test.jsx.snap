// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorControls component renders ColorControls 1`] = `
<DocumentFragment>
  <label
    for="bgColor"
  >
    Background Color
  </label>
  <input
    id="bgColor"
    name="bgColor"
    type="color"
    value="#FF0000"
  />
  <label
    for="fgColor"
  >
    Foreground Color
  </label>
  <input
    id="fgColor"
    name="fgColor"
    type="color"
    value="#0000FF"
  />
  <label
    for="text"
  >
    Text
  </label>
  <input
    id="text"
    name="text"
    type="text"
    value="Hello!"
  />
</DocumentFragment>
`;
